### PR TITLE
Separate load test from depending on network runner

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR separates the load test logic from depending on the network runner, so that the load test can be run by manually specifying a set of timestampvm client URIs to use.

### CODE REVIEW

**Comment by aaronbuchwald** [comment]
Hmm the structure of the load test is a bit broken since canceling the context when we hit the terminal height (which indicates success) can lead to an error in any of the goroutines that are proposing blocks. Will need to restructure this.

File: scripts/versions.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+# Don't export them as they're used in the context of other calls
+avalanche_version=${AVALANCHE_VERSION:-'v1.9.3'}
```
Comment (aaronbuchwald) [inline scripts/versions.sh:4]: nit add new line

File: tests/load/load.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+func newLoadWorker(uri string) *timestampvmLoadWorker {
+	return &timestampvmLoadWorker{
+		uri:    uri,
+		Client: client.New(uri),
+	}
+}
+
+func (t *timestampvmLoadWorker) Name() string {
+	return t.uri
+}
```
Comment (aaronbuchwald) [inline tests/load/load.go:61]: more detailed name

File: tests/load/load_test.go [hunk 3]
```diff
@@ -257,25 +263,16 @@ done:
...
-		u := uris[i] + fmt.Sprintf("/ext/bc/%s", blockchainID)
-		instances[i] = instance{
-			uri: u,
-			cli: client.New(u),
-		}
-	}
+	clientURIs = uris
 
-	defer outf("{{magenta}}logs dir:{{/}} %s\n", logsDir)
+	outf("{{magenta}}logs dir:{{/}} %s\n", logsDir)
```
Comment (aaronbuchwald) [inline tests/load/load_test.go:268]: TODO: separate network start into helper, this is duplicated in e2e test as well

**Review by darioush — APPROVED** [review]

File: scripts/run.sh [hunk 4]
```diff
@@ -39,40 +38,40 @@ fi
 AVALANCHE_LOG_LEVEL=${AVALANCHE_LOG_LEVEL:-INFO}
 
 echo "Running with:"
-echo VERSION: ${VERSION}
+echo avalanche_version: ${avalanche_version}
```
Comment (gyuho) [inline scripts/run.sh:41]: nice!

**Review by gyuho — APPROVED** [review]
lgtm thanks!

> canceling the context when we hit the terminal height (which indicates success) can lead to an error in any of the goroutines that are proposing blocks

I think it's fine, as long as it can exit >terminalHeight? I noticed we also have similar issues in simulator when we cancel the context.

**Comment by aaronbuchwald** [comment]
> lgtm thanks!
> 
> > canceling the context when we hit the terminal height (which indicates success) can lead to an error in any of the goroutines that are proposing blocks
> 
> I think it's fine, as long as it can exit >terminalHeight? I noticed we also have similar issues in simulator when we cancel the context.

Yup same issue is mentioned here: https://github.com/ava-labs/subnet-evm/issues/318.

I think that it's reasonable to have the load test exit gracefully instead of exiting with an error because reaching the terminal height causes the context to be cancelled. I added the quit channel in, so that we have two ways to shut down the worker threads

- cancel and indicate an error occurred
- close and indicate it's time to stop working because the test completed successfully

**Pull request merged**